File: realmate_challenge/models/messages.py

from django.db import models
import uuid
from .conversations import Conversations
class MessageStatusEnum(models.TextChoices):
    RECEIVED = 'RECEIVED'
    SENT = 'SENT'

class Messages(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4)
    direction = models.CharField(
            max_length=9,
            choices=MessageStatusEnum.choices,
            default=MessageStatusEnum.SENT,
        )
    content= models.TextField()
    conversation=models.ForeignKey(Conversations,on_delete=models.RESTRICT)#ninguém disse a regra. sempre assumir menos risco.
    timestamp=models.DateTimeField()

File: realmate-challenge-front-end/types/Messages.ts

export interface Message {
    id: string;             // Unique identifier for the message
    direction: 'RECEIVED' | 'SENT';  // Direction of the message
    content: string;        // Content of the message
    conversation: string;   // The conversation ID this message belongs to
    timestamp: string;      // The timestamp of the message
  }

